trio/_core/_tests/test_run.py

```diff
@@ -1,18 +1,14 @@
 import contextvars
 import functools
-import platform
 import sys
 import threading
 import time
 import types
-import warnings
 import weakref
-from contextlib import contextmanager, ExitStack
+from contextlib import ExitStack
 from math import inf
-from textwrap import dedent
 import gc
 
-import attr
```

Comment:
All the removed imports makes me wonder if this happens for any other test files! Would be neat to clean those up in another PR, IMO.

Reply:
Oh for sure, imports are kind of a mess in this project - which is kind of bound to happen without `isort` or any linter checking for it. See https://github.com/python-trio/trio/pull/2538

```bash
$ flake8 trio/_tests trio/_core/_tests --select F401 | wc -l
45
```
Some of them are false alarms, but vast majority aren't. There's a bunch in non-test files as well, but you'd need rules to exclude the ones used for exporting public symbols.

Reply:
Oh, `F401` is explicitly disabled in check.sh
Well, I'll just make a PR that does per-file disables instead.